### ISSUE TITLE
Developing a helper service that reuses Element

### ISSUE THREAD

**Issue by techmindful** [vector-im/element-web#18688]
Hello. I'm trying to develop a helper service for Matrix and Element. The helper service registers a Matrix account for the user, associates it with a link, and sends the link back to the user. The user can bookmark the link. Whenever the user visits that link, the helper service finds the account from the link's URL parameters, feed the account credentials to the homeserver to retrieve the account's data, and send the data back to the user. This saves the user from installing an app, creating an account, and remembering passwords, when onboarding Matrix.

The problem is, I don't want to write a new Matrix client, or fork Element. When the helper service got the account credentials, or account data (rooms and settings), is there any way for the helper service to feed the account data into Element, so that Element can directly display the stuff that comes after the usual user login?

In other words, typically, user first enters their username and password into Element. Element then sends those to the homeserver, retrieves the access token and account data like rooms and settings, and renders them accordingly. Now in the case with my helper service, is it possible for the helper service to handle the fetching of account data, and feed it into Element, for it to then render?

**Comment by SimonBrandner** [comment]
This is more of a question rather than a bug report/feature request. It would preferable if you could please ask in the [#element-dev room](https://matrix.to/#/#element-dev:matrix.org) or in the [Element Web/Desktop room](https://matrix.to/#/#element-web:matrix.org)
